src/Microsoft.AspNet.Hosting/Internal/RequestServicesContainerMiddleware.cs

```diff
@@ -21,9 +23,37 @@ public RequestServicesContainerMiddleware(RequestDelegate next, IServiceProvider
 
         public async Task Invoke(HttpContext httpContext)
         {
-            using (var container = RequestServicesContainer.EnsureRequestServices(httpContext, _services))
+            // All done if we already have a request services
+            if (httpContext.RequestServices != null)
             {
-                await _next.Invoke(httpContext);
+                return;
+            }
+
+            // Resolve the ScopeFactory from the correct SP
+            var priorApplicationServices = httpContext.ApplicationServices;
```

Comment:
Tweaked the logic a bit to return the applicationServices to httpContext's ApplicationServices which may be null (instead of replacing it with _services)